app/scripts/migrations/151.test.ts

```diff
@@ -0,0 +1,303 @@
+    };
+
+    const newStorage = await migrate(oldStorage);
+    expect(newStorage.data.PermissionController).toStrictEqual({
+      subjects: {
+        [MOCK_SNAP_ID]: {
+          permissions: {
+            [SnapEndowments.EthereumProvider]: {
+              caveats: [],
+              date: 1664187844588,
+              id: 'izn0WGUO8cvq_jqvLQuQP',
+              invoker: MOCK_ORIGIN,
+              parentCapability: SnapEndowments.EthereumProvider,
+            },
+            [Caip25EndowmentPermissionName]: {
+              caveats: [
+                {
+                  type: Caip25CaveatType,
+                  value: {
+                    isMultichainOrigin: false,
+                    optionalScopes: {
+                      'eip155:59141': {
+                        accounts: [],
+                      },
+                    },
+                    requiredScopes: {},
+                  },
+                },
+              ],
+              date: 1723635247705,
+              id: expect.any(String),
+              invoker: MOCK_SNAP_ID,
+              parentCapability: Caip25EndowmentPermissionName,
+            },
+          },
+        },
+      },
+    });
+
+    expect(newStorage.data.SelectedNetworkController).toStrictEqual({
+      domains: {
+        [METAMASK_DOMAIN]: 'mainnet',
+        [MOCK_SNAP_ID]: '0xe705',
+      },
+    });
+  });
+
+  it('merges with an existing permission if the Snap already has `endowment:caip25`', async () => {
+    const oldStorage = {
+      meta: { version: oldVersion },
+      data: {
+        NetworkController: {
+          selectedNetworkClientId: 'mainnet',
+          networkConfigurationsByChainId: {
+            '0x1': {
+              rpcEndpoints: [
+                {
+                  networkClient: 'mainnet',
+                },
+              ],
+            },
+            '0xe705': {
+              rpcEndpoints: [
+                {
+                  networkClient: 'linea-sepolia',
+                },
+              ],
+            },
+          },
+        },
+
+        PermissionController: {
+          subjects: {
+            [MOCK_SNAP_ID]: {
+              permissions: {
+                [SnapEndowments.EthereumProvider]: {
+                  caveats: [],
+                  date: 1664187844588,
+                  id: 'izn0WGUO8cvq_jqvLQuQP',
+                  invoker: MOCK_ORIGIN,
+                  parentCapability: SnapEndowments.EthereumProvider,
+                },
+                [Caip25EndowmentPermissionName]: {
+                  caveats: [
+                    {
+                      type: Caip25CaveatType,
+                      value: {
+                        isMultichainOrigin: false,
+                        optionalScopes: {
+                          'eip155:59141': {
+                            accounts: [
+                              '0x1234567890123456789012345678901234567890',
+                            ],
+                          },
```

Comment:
Just to be realistic, prior to this migration the only scope that this caveat would have is `wallet:eip155` and no accounts
```suggestion
                          'waillet:eip155': {
                            accounts: [],
                          },
```